src/routes/components/useExportLogStream.js

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useContext } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useQuery } from 'react-query';
+
+import { CalloutContext, useOkapiKy } from '@folio/stripes/core';
+
+import downloadBlob from '../../util/downloadBlob';
+
+const useExportLogStream = (job, type) => {
+  const callout = useContext(CalloutContext);
+  const ky = useOkapiKy();
+  const { refetch, isLoading } = useQuery(
+    ['ui-local-kb-admin', 'Logs', 'fetchLogStream', job?.id, type],
+    () => ky(
+      `erm/jobs/${job?.id}/${type}LogStream`
+    ).blob()
+      .then(downloadBlob(`${job?.name}:${type}`)),
+    {
+      // Ensure this doesn't run until the user clicks the button
+      enabled: false
+    }
+  );
+
+  const refetchWithCallout = useCallback(() => {
+    callout.sendCallout({ message: <FormattedMessage id="ui-local-kb-admin.job.log.export.creation" /> });
+    refetch();
+  }, [callout, refetch]);
+
+  if (!job || !type) {
+    return {};
+  }
+
+  return { refetch: refetchWithCallout, isLoading };
+};
+
+export default useExportLogStream;
```

Comment:
Lets export this file from a hooks folder at the same level as the components folder

Reply:
Done